### PR TITLE
✨ Source Google Search Console: Enable direct service account authentication

### PR DESCRIPTION
**Pull request by ad-m**
## What
*Describe what the change is solving*
*It helps to add screenshots if it affects the frontend.*

Add authentication to Google Search Console connector via service account without domain-wide authentication

## How
*Describe the solution*

Currently Google Cloud Search connector supports authentication:

* using OAuth 2.0, which is not optimal for long-term integration (the employee who performs the authorization may leave, etc.)
* using a Service Account and then impersonating the administrator's email address, which requires domain-wide delegation

The current solution allows authentication using a service-account directly. Google Cloud Search allows you to authorize a service account for direct access, so access is possible without impersonation and domain-wide delegation.

It's worth noting that domain-wide delegation is risky permission because it doesn't allow you to specify what service-account user account can access. In other words, it has access to all users in the organization, which in most cases will be over-privileged.

## Recommended reading order
1. `airbyte-integrations/connectors/source-google-search-console/source_google_search_console/service_account_authenticator.py`
2. `airbyte-integrations/connectors/source-google-search-console/source_google_search_console/source.py`

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*

No breaking changes

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- Connector version is set to `0.0.1`
    - `Dockerfile` has version `0.0.1`
- Documentation updated
    - Connector's `README.md`
    - Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - `docs/integrations/README.md`

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Connector Generator</strong></summary>

- Issue acceptance criteria met
- PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests#pull-request-title-convention)
- If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
@ad-m can you make sure all contributors signed the CLA?

**Comment by marcosmarxm** [comment]
@ad-m it looks you contributed with 2 different users. Can you fix the commits?

**Comment by ad-m** [comment]
@marcosmarxm thanks for pointing it out, I was a bit confused as to why SLA is not green yet. All fixed & rebased now.

File: docs/integrations/sources/google-search-console.md [hunk 1]
```diff
@@ -185,11 +186,12 @@ This connector attempts to back off gracefully when it hits Reports API's rate l
...
-| `1.3.0`  | 2023-08-24 | [29750](https://github.com/airbytehq/airbyte/pull/29750)                                                      | Add new `Keyword-Site-Report-By-Site` stream |
-| `1.2.2`  | 2023-08-23 | [29741](https://github.com/airbytehq/airbyte/pull/29741)                                                      | Handle `HTTP-401`, `HTTP-403` errors |
+|:---------|:-----------|:--------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------|
+| `1.3.3`  | 2023-09-04 | [29242](https://github.com/airbytehq/airbyte/pull/29242/files)                                                | Enable direct service account authentication (without domainw-wide delegation)                                                 |
+| `1.3.2`  | 2023-08-25 | [29829](https://github.com/airbytehq/airbyte/pull/29829)                                                      | Make `Start Date` a non-required, added the `suggested streams`, corrected public docs                                         |
```
Comment (ad-m) [inline docs/integrations/sources/google-search-console.md:191]: Whitespace diff

File: docs/integrations/sources/google-search-console.md [hunk 2]
```diff
@@ -185,11 +186,12 @@ This connector attempts to back off gracefully when it hits Reports API's rate l
...
-| `1.3.1`  | 2023-08-24 | [29329](https://github.com/airbytehq/airbyte/pull/29329)                                                      | Update tooltip descriptions                     |
-| `1.3.0`  | 2023-08-24 | [29750](https://github.com/airbytehq/airbyte/pull/29750)                                                      | Add new `Keyword-Site-Report-By-Site` stream |
-| `1.2.2`  | 2023-08-23 | [29741](https://github.com/airbytehq/airbyte/pull/29741)                                                      | Handle `HTTP-401`, `HTTP-403` errors |
+|:---------|:-----------|:--------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------|
+| `1.3.3`  | 2023-09-04 | [29242](https://github.com/airbytehq/airbyte/pull/29242/files)                                                | Enable direct service account authentication (without domainw-wide delegation)                                                 |
```
Comment (ad-m-ss) [inline docs/integrations/sources/google-search-console.md:190]: New entry

**Comment by marcosmarxm** [comment]
Hello @ad-m I'm closing the contribution because of the conflicts, missing issue and (our fault) lack of updates. Please reopen a new pull request if you want to continue the work I'll prioritize it.

**Pull request closed without merging**